### PR TITLE
[BTRX-800] Consent doc answer

### PR DESCRIPTION
**Pull request by tadeork**
## Addresses
https://broadinstitute.atlassian.net/browse/BTRX-800

## Changes
Displays answer if there's no Consent Document uploaded when the CG was created.
Validates the form when is and is not present the Consent Document.

## Testing
When creating a new Consent Group don't upload Consent Document and answer the question.
At review time of the created Consent Group upload a Consent Group, then go back to the tab of Details: the answer should be gone.
Remove the document: in the review tab the field of the answer should appear empty.

---

- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Get a thumb from Belatrix
- [ ] **Submitter**: Get a thumb from @rushtong
- [ ] **Submitter**: Merge to develop using github's "Squash and Merge" feature
- [ ] **Submitter**: Delete branch after merge

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -72,6 +72,8 @@
     "react-router-dom": "5.0.1",
     "react-select": "2.1.2",
     "regenerator-runtime": "0.13.2",
-    "xlsx": "0.14.3"
+    "rxjs": "^6.5.3",
```
Comment (verovicario) [inline package.json]: remove ^

File: grails-app/controllers/org/broadinstitute/orsp/api/NewConsentGroupController.groovy [hunk 2]
```diff
@@ -130,6 +130,13 @@ class NewConsentGroupController extends AuthenticatedController {
         render([message: 'Consent Group was updated'] as JSON)
     }
 
+    def deleteNoConsentReason() {
+        Issue issue = Issue.findByProjectKey(params.consentKey)
```
Comment (verovicario) [inline grails-app/controllers/org/broadinstitute/orsp/api/NewConsentGroupController.groovy:134]: Avoid using GORM

File: src/main/webapp/components/Documents.js [hunk 3]
```diff
@@ -241,7 +241,8 @@ export const Documents = hh(class Documents extends Component {
         user: this.props.user,
         handleLoadDocuments: this.props.handleLoadDocuments,
         userName: this.props.userName,
-        isConsentGroup: this.props.isConsentGroup
+        isConsentGroup: this.props.isConsentGroup,
+        deleteNoConsentReason: this.props.deleteNoConsentReason,
```
Comment (verovicario) [inline src/main/webapp/components/Documents.js]: remove comma

**Review by verovicario — APPROVED** [review]

**Review by LeoForconesi — APPROVED** [review]
:+1:

File: grails-app/controllers/org/broadinstitute/orsp/api/NewConsentGroupController.groovy (hunk 2 above)
Comment (rushtong) [reply grails-app/controllers/org/broadinstitute/orsp/api/NewConsentGroupController.groovy:134]: 👍 to this suggestion. The conversation is marked as resolved, but the suggestion was not implemented.

File: grails-app/controllers/org/broadinstitute/orsp/api/NewConsentGroupController.groovy [hunk 4]
```diff
@@ -130,6 +130,13 @@ class NewConsentGroupController extends AuthenticatedController {
...
+        issueService.removeNoConsentAnswer(issue)
+        response.status = 200
+        render([message: 'Deleted No Consent Form reason'] as JSON)
```
Comment (rushtong) [inline grails-app/controllers/org/broadinstitute/orsp/api/NewConsentGroupController.groovy:137]: APIs, in general, should return the thing that is modified. In this case, it should be returning the modified issue.

**Pull request merged**